spec/requests/private_items_controller_spec.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe PrivateItemsController, type: :request do
+  describe 'GET#index' do
+    context 'when the user is not signed in' do
+      it 'redirects to the new user session page' do
+        get private_items_path
+
+        expect(response).to have_http_status(:found)
+        expect(response).to redirect_to(new_user_session_path)
```

Comment:
For the API, i think we don't need to assert the `redirect_to` path. Because we just need the status for the client process the next step right?

```suggestion
```

Reply:
I agree this is unnecessary, it will be removed when clean up none-API paths 🙏